tools/networkmanager/inject_txs.go

```diff
@@ -19,19 +21,49 @@ import (
 	"github.com/obscuronet/obscuro-playground/integration/simulation"
 )
 
-func InjectTransactions(nmConfig Config) {
+func InjectTransactions(cfg Config) {
 	hostConfig := config.HostConfig{
-		L1NodeHost:          nmConfig.l1NodeHost,
-		L1NodeWebsocketPort: nmConfig.l1NodeWebsocketPort,
-		L1ConnectionTimeout: nmConfig.l1ConnectionTimeout,
+		L1NodeHost:          cfg.l1NodeHost,
+		L1NodeWebsocketPort: cfg.l1NodeWebsocketPort,
+		L1ConnectionTimeout: cfg.l1ConnectionTimeout,
 	}
 	l1Client, err := ethclient.NewEthClient(hostConfig)
 	if err != nil {
 		panic(fmt.Sprintf("could not create L1 client. Cause: %s", err))
 	}
-	l2Client := obscuroclient.NewClient(nmConfig.obscuroClientAddress)
+	l2Client := obscuroclient.NewClient(cfg.obscuroClientAddress)
 
+	txInjector := simulation.NewTransactionInjector(
+		1*time.Second,
+		stats.NewStats(1),
+		[]ethclient.EthClient{l1Client},
+		createWallets(cfg, l1Client, l2Client),
+		&cfg.mgmtContractAddress,
+		[]obscuroclient.Client{l2Client},
+		mgmtcontractlib.NewMgmtContractLib(&cfg.mgmtContractAddress),
+		erc20contractlib.NewERC20ContractLib(&cfg.mgmtContractAddress, &cfg.erc20ContractAddress),
+	)
+
+	// We listen for interrupts, to log statistics before exiting.
```

Comment:
Maybe this method could do with a high level comment somewhere that says what it does. I'm confused, it looks like it's stopping injecting transactions before it starts injecting them...

Reply:
Oh I'm being dense, this just listens for the interrupt, doesn't immediately stop.